### PR TITLE
Implement sparklyr changes for Databricks notebook-scoped libraries

### PR DESCRIPTION
**Pull request by jlee2000**
I am creating this PR to give sparklyr users a better experience on Databricks with respect to notebook-scoped libraries.

### CODE REVIEW

File: R/databricks_connection.R [hunk 1]
```diff
@@ -40,6 +40,11 @@ databricks_connection <- function(config, extensions) {
     }
   )
 
+  # Hand in driver's libPaths to worker if user hasn't already set libpaths for notebook-scoped libraries
```
Comment (falaki) [inline R/databricks_connection.R:43]: Can we add a unit test for this in https://github.com/sparklyr/sparklyr/blob/main/tests/testthat/test-databricks-connect.R?

File: R/databricks_connection.R [hunk 2]
```diff
@@ -40,6 +40,11 @@ databricks_connection <- function(config, extensions) {
...
+  if (is.null(config$spark.r.libpaths)) {
+    config$spark.r.libpaths <- toString(.libPaths())
```
[the remark this answers is not in the document]
Comment (yitao-li) [reply R/databricks_connection.R:45]: We also typically shorten the `if (is.null(x)) x <- expr` type of assignment into `x <- x %||% expr` in `sparklyr` code base using the `%||%` operator from `rlang`.

**Review by yitao-li — CHANGES_REQUESTED** [review]
Looks good. :+1: Thanks for contributing to `sparklyr`!

- I hope this will address some pain points with Databricks use cases. Do you want to also update the `NEWS.md` file outlining the effect of this change, and also (maybe) briefly the rationale behind it?

- Also a question I have: does this make sense for all Databricks use cases? Are directory layouts on Spark driver and Spark worker noes guaranteed to be identical for notebook-scoped libs?

- Please see inline comment for suggestions on `toString()` and coding style in R.

File: R/databricks_connection.R [hunk 3]
```diff
@@ -40,6 +40,11 @@ databricks_connection <- function(config, extensions) {
...
+  }
```
Comment (yitao-li) [inline R/databricks_connection.R:46]:
```suggestion
    config$spark.r.libpaths <- config$spark.r.libpaths %||% paste(.libPaths(), collapse = ",")
```

Comment (yitao-li) [reply R/databricks_connection.R:46]: We also typically shorten the

```
if (is.null(x))
  x <- expr
```
type of assignment into `x <- x %||% expr` in sparklyr code base using the `%||%` operator from rlang.

**Review by yitao-li — CHANGES_REQUESTED** [review]

File: R/databricks_connection.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (jlee2000) [reply R/databricks_connection.R:45]: You're right, that works much better.

**Comment by jlee2000** [comment]
@yitao-li Thank you for the comments! They were very helpful. To answer your question: yes, this makes sense for Databricks use cases. The directory layout for workers and drivers is guaranteed to be the same.

File: NEWS.md [hunk 4]
```diff
@@ -76,6 +76,12 @@
 - Fixed a bug that previously caused invoke params containing `NaN`s to be
   serialized incorrectly.
 
+### Connections
+
+- Added support for notebook-scoped libraries on Databricks connections.
```
Comment (falaki) [inline NEWS.md:81]:
```suggestion
- Added support for notebook-scoped libraries on Databricks.
```

**Review by falaki — APPROVED** [review]

**Review by yitao-li — APPROVED** [review]
LGTM :slightly_smiling_face:

**Pull request merged**